### DISCUSSION TITLE
How a user can update own profile - updateUser

### DISCUSSION THREAD

**Discussion by beactivesolutions** [strapi/strapi#8096] [Legacy Topics]
In the sample GraphQL query, if I change the ID, I can change the data of each user.
How can I restrict it to my account only?

`mutation{
  updateUser(input: {
    where:{
      id:5
    },
    data: {
      age:54
    }
  }){
    user{
      id
      age
    }
  }
}`

**Comment by derrickmehaffy** [comment]
Hi @beactivesolutions I can provide an answer but we have started to decommission the usage of GitHub Discussions and are moving over to our forum, in the future can you please ask questions on https://forum.strapi.io/c/support/7

In this case, much like REST, you will want to use a policy. You can read more about policies and how to use them with GraphQL here:
https://strapi.io/documentation/v3.x/concepts/policies.html
https://strapi.io/documentation/v3.x/plugins/graphql.html#execute-a-policy-before-a-resolver

**Reply by derrickmehaffy** [reply]
Locking discussion as user posted on forum: https://forum.strapi.io/t/how-a-user-can-update-own-profile-updateuser/197/2
